slides/src/content/ngrx.md

```diff
@@ -0,0 +1,171 @@
+# Redux (@ngrx)
+
+---
+
+## Motivation 
+
+1. Create [Single Source of Truth](https://en.wikipedia.org/wiki/Single_source_of_truth) 
+1. [Separation of Concern](https://en.wikipedia.org/wiki/Separation_of_concerns) : Separation of Data and Logic
+1. Improve scalability by having things _react_ to events as opposed to caller dictating actions
+1. Reduce Complexity of Mental Map by promoting separation of responsibility between components
+1. Create a coherent framework so future developers will know how to add features by following convention
+1. Uni-Directional Data Flow : Easy to reason about the interractions between actors
+
+---
+
+## Structure of Systems - Redux vs Server Architecture
+
+| Function | Angular With @ngrx  | Server |
+|---|---|---|
+| Data Storage | Store  |  Database  | 
+| Presentation View |  @Component  | Client API  |
+| Logic to massage data for Storage |  @Effects  | Services |
+| Communication | Action / Observables  | Function Invocation | 
+
+---
+
+## Store (1/2)
+
+1. Responsible for application state (Immutability, Persisting to Disk etc.)
+1. The [Single Source of Truth](https://en.wikipedia.org/wiki/Single_source_of_truth)
+1. Keeper of stateful information, void of business logic
+1. Can only be affected by actions
+1. Prevent unintentional modifications by other methods (object reference, Direct access)
+
+---
+
+## Sample Store (2/2)
+
+```ts
+//Root module
+import { StoreModule } from '@ngrx/store';
+import { people } from './people'; // Reducers
+
+@NgModule({
+  imports:      [ ..., StoreModule.provideStore({ people }) ],
+})
+export class AppModule {  }
+```
+
+```ts
+//Reducer
+export const people = (state = [], action) => {
+  switch (action.type) {
+    case PeopleActions.REMOVE_PERSON:
+      return state.filter(person => person.id !== action.payload);
+    default:
+      return state;
+  }
+}
+```
+
+---
+
+## @Effects (1/2)
+
+1. Responsible for Business Logic and Async actions (Http Calls)
+1. Does not keep local state
+1. Listens on the Action Stream and adheres to 'Action In Action Out'
+1. Typical use is take user input, make http call, and provide output to go into store
+
+---
+
+## @Effects Code (2/2)
```

Comment:
Because we have a limited vertical space, I try avoiding using things like this `1/2` or `2/2`. Instead it's better in my opinion to add something more meaningful like "Handling Side Effects with `@Effect`". 

The previous slide title could be: "The `ngrx/effects` Library"